src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/WMSCascadeTest.java

```diff
@@ -189,13 +188,54 @@ public void testCascadeGetLegendRequestJSON() throws Exception {
 
         print(dom);
         JSONObject responseJson = JSONObject.fromObject(dom.toString());
-        assertTrue(responseJson.has(JSONLegendGraphicBuilder.LEGEND));
+        // styless remote layer..fixed wrong assertion
+        assertTrue(responseJson.isEmpty());
+    }
 
-        JSONArray legendArray = responseJson.getJSONArray(JSONLegendGraphicBuilder.LEGEND);
-        assertFalse(legendArray.isEmpty());
+    @Test
+    public void testScaleDenominator() throws Exception {
+        LayerGroupInfo info = getCatalog().getLayerGroupByName("cascaded_group");
+        LayerInfo groupLayer2 = getCatalog().getLayerByName("group_lyr_2");
+        assertNotNull(info);
+        assertNotNull(groupLayer2);
+        ReferencedEnvelope request1 =
+                new ReferencedEnvelope(groupLayer2.getResource().getNativeBoundingBox());
+        // minx,miny,maxx,maxy
+        String lyrBBox =
+                request1.getMinX()
+                        + ","
+                        + request1.getMinY()
+                        + ","
+                        + request1.getMaxX()
+                        + ","
+                        + request1.getMaxY();
+
+        // configure max scale on one of the layers
+        // set max scale as small as possible to have it filtered
+        WMSLayerInfo groupLayer1WMSResource =
+                (WMSLayerInfo) getCatalog().getLayerByName("group_lyr_1").getResource();
+        groupLayer1WMSResource.setMinScale(1);
```

Comment:
Set and reset best put in a try/finally block, otherwise a failure in this test can cause spurious failures in other tests.

Reply:
understood, will do